lib/Parse/ParseType.cpp

```diff
@@ -1181,10 +1181,17 @@ Parser::parseTypeOptional(TypeRepr *base) {
   llvm::Optional<TypeSyntax> SyntaxNode;
   if (SyntaxContext->isEnabled()) {
     OptionalTypeSyntaxBuilder Builder(Context.getSyntaxArena());
-    Builder
-      .useQuestionMark(SyntaxContext->popToken())
-      .useWrappedType(SyntaxContext->popIf<TypeSyntax>().getValue());
-    SyntaxNode.emplace(Builder.build());
+    auto QuestionMark = SyntaxContext->popToken();
+    auto WrappedType = SyntaxContext->popIf<TypeSyntax>();
+    if (WrappedType.hasValue()) {
```

Comment:
```cpp
if (auto WrappedType = SyntaxContext->popIf<TypeSyntax>()) {
```
should work.

Reply:
Oh that's great. I always thought that the `if (auto …)` syntax only worked for pointers.